pkg/config/context.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	"context"
+	PermitErrors "github.com/permitio/permit-golang/pkg/errors"
+	"github.com/permitio/permit-golang/pkg/openapi"
+	"strings"
+)
+
+type PermitContext struct {
+	APIKeyLevel   APIKeyLevel
+	ProjectId     string
+	EnvironmentId string
+}
+
+type PermitContextInterface interface {
+	GetProjectId() string
+	GetEnvironmentId() string
+	SetContext(project string, environment string)
+	GetContext() *PermitContext
+}
+
+func (p *PermitContext) SetContext(project string, environment string, apiKeyLevel APIKeyLevel) {
+	p.ProjectId = project
+	p.EnvironmentId = environment
+	p.APIKeyLevel = apiKeyLevel
+
+}
+
+func (p *PermitContext) GetContext() *PermitContext {
+	return p
+}
+
+func (p *PermitContext) GetEnvironment() string {
+	return p.EnvironmentId
+}
+
+func (p *PermitContext) GetProject() string {
+	return p.ProjectId
+}
+
+func PermitContextFactory(ctx context.Context, client *openapi.APIClient, project string, environment string, isUserInput bool) (*PermitContext, error) {
+	apiKeysScopeRead, httpRes, err := client.APIKeysApi.GetApiKeyScope(ctx).Execute()
+	err = PermitErrors.HttpErrorHandle(err, httpRes)
+	if err != nil {
+		if strings.Contains(err.Error(), string(PermitErrors.ForbiddenMessage)) {
```

Comment:
`NewPermitContextError(err.Error())` this will lead to recursive error and it will look bad as the error string contain other arguments (such as errorType).
Made it more elegant though.